File: python/collatz-conjecture/collatz_conjecture.py
def steps(number):
    if (number < 1):
        raise ValueError("Only positive integers are allowed")
    
    if number == 1:
        return 0

    if number % 2 == 0:
        return 1 + steps(number // 2)    
    else:
        return 1 + steps(3 * number + 1)

File: javascript/collatz-conjecture/collatz-conjecture.js
export const steps = (number) => {
    if (number < 1) {
        throw new Error('Only positive numbers are allowed');
    }

    if (number == 1) {
        return 0;
    }

    number = number % 2 == 0 ? number / 2 : 3 * number + 1

    return 1 + steps(number);
};
